web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -350,31 +350,30 @@ export function Navigation() {
 
     // If the page is not part of the sidenav, such as Account Settings, curentPageSection will be undefined, and the drawer should be collapsed.
     if (currentPageSection) {
+      // If there is already an expanded section set, don't change it.
+      if (!!debouncedSection) {
+        return;
+      }
       handleSetExpandedSection(currentPageSection);
     } else {
       collapseDrawer(false);
     }
   }, [currentPageSection]);
 
-  // Handler for navigation actions
-  const handleNavigation = useCallback(
-    (route: string) => {
-      history.push(route);
```

Comment:
This `history.push` was unnecessary since `SubsectionItem`s are already `NavLink`s, meaning they already have native navigation functionality